test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,558 @@
+						b_height: 1,
+						t_id: '3634383815892709955',
+						t_type: 2,
+					},
+					{
+						b_id: '6524861224470851000',
+						b_height: 1,
+						t_id: '3634383815892709956',
+						t_type: 1,
+					},
+					{
+						b_id: '6524861224470851000',
+						b_height: 1,
+						t_id: '3634383815892709957',
+						t_type: 0,
+					},
+				];
+
+				library.db.blocks.loadLastBlock = sinonSandbox
+					.stub()
+					.resolves(genesisBlock_votes);
+
+				blocksUtilsModule.loadLastBlock((err, block) => {
+					expect(block).to.be.an('object');
+					expect(block.id).to.equal('6524861224470851000');
+					expect(block.transactions).to.be.an('array');
+					expect(block.transactions[0]).to.be.an('object');
+					expect(block.transactions[0].id).to.equal('1465651642158264047');
+					expect(block.transactions[0].type).to.equal(3);
+					expect(block.transactions[1]).to.be.an('object');
+					expect(block.transactions[1].id).to.equal('3634383815892709955');
+					expect(block.transactions[1].type).to.equal(2);
+					expect(block.transactions[2]).to.be.an('object');
+					expect(block.transactions[2].id).to.equal('3634383815892709957');
+					expect(block.transactions[2].type).to.equal(0);
+					expect(block.transactions[3]).to.be.an('object');
+					expect(block.transactions[3].id).to.equal('3634383815892709956');
+					expect(block.transactions[3].type).to.equal(1);
+					done();
 				});
 			});
 
-			it('should call modules.blocks.lastBlock.set with block');
+			it('should set the last block', done => {
+				library.db.blocks.loadLastBlock = sinonSandbox
+					.stub()
+					.resolves(viewRow_full_blocks_list);
 
-			it('should call callback with error = null');
+				modules.blocks.lastBlock.set = sinonSandbox.spy();
 
-			it('should call callback with result containing the block');
+				blocksUtilsModule.loadLastBlock((err, block) => {
+					expect(block).to.be.an('object');
+					expect(block.id).to.equal('13068833527549895884');
+					expect(modules.blocks.lastBlock.set).to.have.been.calledWith(block);
+					done();
+				});
+			});
 		});
 	});
 
 	describe('getIdSequence', () => {
-		it('should call modules.blocks.lastBlock.get');
-
-		it('should call library.db.query');
-
-		it('should call library.db.query with sql.getIdSequence');
-
-		it(
-			'should call library.db.query with {height: height, limit: 5, delegates: constants.activeDelegates}'
-		);
-
-		describe('when db query fails', () => {
-			it('should call logger.error with error stack');
-
-			it('should call callback with the Blocks#getIdSequence error');
-		});
-
-		describe('when db query succeeds', () => {
-			describe('and returns no results', () => {
-				it('should call callback with an error');
+		it('should return error when library.db.blocks.getIdSequence fails', done => {
+			blocksUtilsModule.getIdSequence(10, (err, sequence) => {
+				expect(loggerStub.error.args[0][0]).to.contains(
+					"TypeError: Cannot read property 'length' of undefined"
```

Comment:
why type error?

Reply:
It's the expected error if the promise was to yield a value of undefined.